src/httpsnippet.ts

```diff
@@ -0,0 +1,327 @@
+      }
+    });
+  }
+
+  prepare = (harRequest: HarRequest) => {
+    const request: Request = {
+      ...harRequest,
+      fullUrl: '',
+      uriObj: {} as UrlWithParsedQuery,
+      queryObj: {},
+      headersObj: {},
+      cookiesObj: {},
+      allHeaders: {},
+    };
+
+    // construct query objects
+    if (request.queryString && request.queryString.length) {
+      debug.info('queryString found, constructing queryString pair map');
+
+      request.queryObj = request.queryString.reduce(reducer, {});
+    }
+
+    // construct headers objects
+    if (request.headers && request.headers.length) {
+      const http2VersionRegex = /^HTTP\/2/;
+      request.headersObj = request.headers.reduce((accumulator, { name, value }) => {
+        const headerName = http2VersionRegex.exec(request.httpVersion) ? name.toLocaleLowerCase() : name;
+        return {
+          ...accumulator,
+          [headerName]: value,
+        };
+      }, {});
+    }
+
+    // construct headers objects
+    if (request.cookies && request.cookies.length) {
+      request.cookiesObj = request.cookies.reduceRight(
+        (accumulator, { name, value }) => ({
+          ...accumulator,
+          [name]: value,
+        }),
+        {},
+      );
+    }
+
+    // construct Cookie header
+    const cookies = request.cookies?.map(({ name, value }) => `${encodeURIComponent(name)}=${encodeURIComponent(value)}`);
+
+    if (cookies?.length) {
+      request.allHeaders.cookie = cookies.join('; ');
+    }
+
+    switch (request.postData.mimeType) {
+      case 'multipart/mixed':
+      case 'multipart/related':
+      case 'multipart/form-data':
+      case 'multipart/alternative':
+        // reset values
+        request.postData.text = '';
+        request.postData.mimeType = 'multipart/form-data';
+
+        if (request.postData?.params) {
+          const form = new FormData();
+
+          // The `form-data` module returns one of two things: a native FormData object, or its own polyfill
+          // Since the polyfill does not support the full API of the native FormData object, when this library is running in a browser environment it'll fail on two things:
+          //
+          //  1. The API for `form.append()` has three arguments and the third should only be present when the second is a
+          //    Blob or USVString.
+          //  1. `FormData.pipe()` isn't a function.
+          //
+          // Since the native FormData object is iterable, we easily detect what version of `form-data` we're working with here to allow `multipart/form-data` requests to be compiled under both browser and Node environments.
+          //
+          // This hack is pretty awful but it's the only way we can use this library in the browser as if we code this against just the native FormData object, we can't polyfill that back into Node because Blob and File objects, which something like `formdata-polyfill` requires, don't exist there.
+          // @ts-expect-error TODO
```

Comment:
Yeah... I'm sorry about this. I made an attempt to type this same thing in some stuff I'm working on in an unrelated project and pretty quickly gave up. I wish working with these FormData polyfills wasn't such a disaster or required ESM codebase migrations.

Reply:
nah.  no need to apologize.  I appreciate any-and-all efforts to try to fix this.  We'll definitely need to loop back to this some time after this PR merges.